crates/oxc_query/src/properties.rs

```diff
@@ -0,0 +1,584 @@
+                .local
+                .name
+                .to_string()
+                .into()
+        }),
+        _ => {
+            unreachable!(
+                "attempted to read unexpected property '{property_name}' on type 'DefaultImport'"
+            )
+        }
+    }
+}
+
+pub(super) fn resolve_expression_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    _resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "as_constant_string" => resolve_property_with(contexts, |v| {
+            v.as_constant_string().map_or(FieldValue::Null, Into::into)
+        }),
+        _ => {
+            unreachable!(
+                "attempted to read unexpected property '{property_name}' on type 'Expression'"
+            )
+        }
+    }
+}
+
+pub(super) fn resolve_import_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    _resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "from_path" => resolve_property_with(contexts, |v| {
+            v.as_import()
+                .unwrap_or_else(|| {
+                    panic!("expected to have an import vertex, instead have: {v:#?}")
+                })
+                .import
+                .source
+                .value
+                .to_string()
+                .into()
+        }),
+        _ => {
+            unreachable!("attempted to read unexpected property '{property_name}' on type 'Import'")
+        }
+    }
+}
+
+pub(super) fn resolve_import_ast_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "from_path" => resolve_import_property(contexts, property_name, resolve_info),
+        _ => {
+            unreachable!(
+                "attempted to read unexpected property '{property_name}' on type 'ImportAST'"
+            )
+        }
+    }
+}
+
+pub(super) fn resolve_interface_extend_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    _resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "str" => interface_extend_implem(contexts),
+        _ => {
+            unreachable!(
+                "attempted to read unexpected property '{property_name}' on type 'InterfaceExtend'"
+            )
+        }
+    }
+}
+
+pub(super) fn resolve_jsxattribute_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    _resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "name" => resolve_property_with(contexts, |v| {
+            jsx_attribute_name_to_string(
```

Comment:
Perhaps another good candidate function to convert to `-> FieldValue` instead of `-> String`?